pkg/scheduler/framework/plugins/interpodaffinity/plugin.go

```diff
@@ -132,3 +134,77 @@ func GetNamespaceLabelsSnapshot(logger klog.Logger, ns string, nsLister listersv
 	logger.V(3).Info("getting namespace, assuming empty set of namespace labels", "namespace", ns, "err", err)
 	return
 }
+
+func (pl *InterPodAffinity) isSchedulableAfterPodChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	originalPod, modifiedPod, err := util.As[*v1.Pod](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, err
+	}
+	if (modifiedPod != nil && modifiedPod.Spec.NodeName == "") || (originalPod != nil && originalPod.Spec.NodeName == "") {
+		logger.V(5).Info("the added/updated/deleted pod is unscheduled, so it doesn't make the target pod schedulable",
+			"pod", klog.KObj(pod), "originalPod", klog.KObj(originalPod), "modifiedPod", klog.KObj(modifiedPod))
+		return framework.QueueSkip, nil
+	}
+
+	terms, err := framework.GetAffinityTerms(pod, framework.GetPodAffinityTerms(pod.Spec.Affinity))
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	antiTerms, err := framework.GetAffinityTerms(pod, framework.GetPodAntiAffinityTerms(pod.Spec.Affinity))
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	// Pod is added or updated.
+	// Return Queue when the added/updated pod matching the target pod's affinity or not matching anti-affinity
+	if modifiedPod != nil {
+		if (len(terms) != 0 && !podMatchesAllAffinityTerms(terms, modifiedPod)) ||
+			(len(antiTerms) != 0 && podMatchesAllAffinityTerms(antiTerms, modifiedPod)) {
```

Comment:
IIUC, we should return `Queue` when an incoming pod's label changed from satisfying target pod's anit-affinity to **not** satisfying target pod's anit-affinity. PTAL if you have time. Thanks. @sanposhiho @AxeZhan

Reply:
What do you mean by `incoming pod`, if it's an existing pod which got label changed. Then I think yes, we should return `Queue` in this case.

Reply:
incoming pod means a pod which was added or an existing pod which label was updated

Reply:
I prefer we only care `existing pods` here. Let's wait for sanposhiho.

Reply:
Regarding PodAntiAffinity, I agree with @AxeZhan. Probably we should distinguish Pod/Add and Pod/Update events:
- For PodAffinity, we should return `Queue` when a new scheduled Pod satisfies target pod's affinity **_or_** an existing scheduled Pod got changed from non-match to match the affinity.
- For PodAntiAffinity, we should return `Queue` **_only_** when an existing scheduled Pod got changed from non-match to match the affinity.

Reply:
> * For PodAntiAffinity, we should return `Queue` **_only_** when an existing scheduled Pod got changed from non-match to match the affinity.

I want to confirm that, do you mean that we should return `Queue` only when an existing scheduled pod got changed from **_match to non-match_** the **_anti-affinity_**, but not `from non-match to match the affinity`.

Reply:
Oops, the above message is wrong: you're right, I meant to say:
For PodAntiAffinity, we should return Queue only when an existing scheduled Pod got changed from match to non-match the anti-affinity.